sdl2promela/sdl/model.py

```diff
@@ -1715,6 +1768,73 @@ def __gather_inputs(self):
                         VariableReference(param) for param in input_block.parameters
                     ]
                     trigger.transitions[target] = block
+        self.__gather_observer_inputs()
+
+    def __gather_observer_inputs(self):
+        for state, signals in self.source.cs_mapping.items():
+            for signal in signals:
+                if signal.observer_input is not None:
+                    continue
+                if signal.inputString.startswith(
+                    "input"
+                ) or signal.inputString.startswith("output"):
+                    self.__parse_observer_input(signal, state)
+
+    def __parse_observer_input(self, signal: ogAST.ContinuousSignal, state: str):
+        info, parameter = _parseObserverInput(signal.inputString)
+        self.observer_attachments.append(info)
+        parameter_type = None
+        if info.recipientName is not None:
+            type_name = "{}-Event-msg-in".format(info.recipientName.capitalize())
```

Comment:
We could try to extract this for greater readability (having to handle -in/-out will make it difficult to save lines, but still, the end result will be easier to read IMHO).

Reply:
Refactored